apollo/smoothie.py

```diff
@@ -58,70 +38,194 @@ def connect(self):
+        
+            # Handle G92 GCode
+            if gcode.startswith('G92') and not is_gcode_done:
+                if response == 'ok':
+                    is_gcode_done = True
+
+            # Handle G90
+            if gcode.startswith('G90') and not is_gcode_done:
+                if response == 'ok':
+                    is_gcode_done = True
+
+            # Handle G91
+            if gcode.startswith('G91') and not is_gcode_done:
+                if response == 'ok':
+                    is_gcode_done = True
+
+            # Handle M112
+            if gcode.startswith('M112') and not is_gcode_done:
+                if response.startswith('ok'):
+                    is_gcode_done = True
+
+            # Handle M119
+            if gcode.startswith('M119') and not is_gcode_done:
+                try:
+                    json_result = json.loads(response)
+                    is_gcode_done = True
+                except ValueError:
+                    pass
 
-                # check whether has "stat" and value if so
-                if 'stat' in jobj:
-                    if jobj['stat'] == 0:
-                        done = True
+            # Handle M199
+            if gcode.startswith('M199') and not is_gcode_done:
+                try:
+                    json_result = json.loads(response)
+                    is_gcode_done = True
+                except ValueError:
+                    pass
 
-        if response_handler:
-            response_handler.send(None)
+            # Handle M204
+            if gcode.startswith('M204') and not is_gcode_done:
+                try:
+                    #status, data = response.split(' ')
+                    json_result = json.loads(response)
+                    is_gcode_done = True
+                except ValueError:
+                    pass
 
-        return res
+            # Handle: M999
+            if gcode.startswith('M999') and not is_gcode_done:
+                if response == 'ok':
+                    is_gcode_done = True
+
+            if response == '{"stat":0}':
+                break
+
+        if not (gcode == 'M112'):
+            yield from self.turn_off_feedback()
+
+        return json_result
+
+    @asyncio.coroutine
+    def turn_on_feedback(self):
+        yield from self.send_feedback_gcode('M62\r\n', 'feedback engaged')
+
+    @asyncio.coroutine
+    def turn_off_feedback(self):
+        yield from self.send_feedback_gcode('M63\r\n', 'feedback disengaged')
+
+    @asyncio.coroutine
+    def send_feedback_gcode(self, delimited_gcode, expected_response_msg):
+        
+        yield from self.write_and_drain(delimited_gcode)
+
+        read_attempts = 0
+        while read_attempts < Config.MAX_FEEDBACK_READS:
+            read_attempts += 1
+            try:
+                response = (yield from self._read())
+                logger.info('Smoothie send_feedback_gcode response(1): {}'.format(response))
+            except:
```

Comment:
blanket except statements aren't good practice because they swallow all exceptions. We should know what sort of exceptions our code will be throwing. So preferrably,

`except SomeException as e`

and maybe multiple rows of exception handling is needed...